code/game/machinery/firealarm.dm

```diff
@@ -306,104 +307,180 @@
 /obj/machinery/firealarm/attack_robot_secondary(mob/user)
 	return attack_hand_secondary(user)
 
-/obj/machinery/firealarm/attackby(obj/item/tool, mob/living/user, list/modifiers, list/attack_modifiers)
-	add_fingerprint(user)
-
-	if(tool.tool_behaviour == TOOL_SCREWDRIVER && buildstage == FIRE_ALARM_BUILD_SECURED)
-		tool.play_tool_sound(src)
+/obj/machinery/firealarm/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	if(issilicon(user))
+		context[SCREENTIP_CONTEXT_CTRL_LMB] = "Toggle automatic fire detection"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	if(isnull(held_item))
+		context[SCREENTIP_CONTEXT_LMB] = "Turn on"
+		context[SCREENTIP_CONTEXT_RMB] = "Turn off"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	switch(held_item.tool_behaviour)
+		if(TOOL_SCREWDRIVER)
+			context[SCREENTIP_CONTEXT_RMB] = "[panel_open ? "Unexpose" : "Expose"] wires"
+			. = CONTEXTUAL_SCREENTIP_SET
```

Comment:
this doesn't have a `buildstage` check like some of the other tool contexts, but does care about it?